includes/Core/Modules/Module_With_Settings_Trait.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Trait Google\Site_Kit\Core\Modules\Module_With_Settings_Trait
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Modules;
+
+/**
+ * Trait for a module that includes a screen.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+trait Module_With_Settings_Trait {
+
+	/**
+	 * Settings instance.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @var Module_Settings
+	 */
+	protected $settings;
+
+	/**
+	 * Gets the module-specific settings instance.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Module_Settings
+	 */
+	abstract protected function get_settings_instance();
```

Comment:
Yes, it's the same behavior as an abstract method in parent class - in this case the trait is like the parent and requires the inheriting class ("child") to implement the method (it's not overriding it).
https://www.php.net/manual/en/language.oop5.traits.php#language.oop5.traits.abstract

I haven't checked it outside of Travis, but I'm not aware of any change that would cause the code to be invalid in older versions. Support for traits was added in 5.4 so simply running the tests with this would be sufficient but I haven't done that yet, but I can if you'd like?

Reply:
Ah, learned something new again! :)

That looks good to me then. I think we should rename the method though to `setup_settings()`, to make it consistent with similar cases (`get_client()` --> `setup_client()`, `get_service()` --> `setup_services()`).